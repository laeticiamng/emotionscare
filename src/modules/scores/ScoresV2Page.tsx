"use client";
import React from "react";
import { PageHeader, Card, Button } from "@/COMPONENTS.reg";
import ScoresV2Panel from "@/modules/scores/ScoresV2Panel";

export default function ScoresV2Page() {
  return (
    <main aria-label="Scores & vibes">
      <PageHeader title="Scores & vibes" subtitle="Visualisation temps réel des humeurs et séances" />
      <Card>
        <ScoresV2Panel />
        <div style={{ marginTop: 12 }}>
          <Button href="/app/scores">Accéder à la page principale</Button>
        </div>
      </Card>
    </main>
  );
}
