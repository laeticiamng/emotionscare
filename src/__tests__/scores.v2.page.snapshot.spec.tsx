import React from "react";
import { describe, it, expect, vi } from "vitest";
import { render } from "@testing-library/react";
vi.mock("@/COMPONENTS.reg", () => ({
  PageHeader: ({ title, subtitle }: any) => React.createElement('header', null, React.createElement('h1', null, title), React.createElement('p', null, subtitle)),
  Card: ({ children }: any) => React.createElement('div', null, children),
  Button: (props: any) => React.createElement('a', props),
  ProgressBar: ({ value, max }: any) => React.createElement('progress', { value, max }),
  Sparkline: ({ values }: any) => React.createElement('div', null, values?.join(',')),
  BadgeLevel: ({ level }: any) => React.createElement('span', null, level),
}));
vi.mock("@/modules/scores/ScoresV2Panel", () => ({ default: () => React.createElement('div', null, 'ScoresV2Panel') }));
import ScoresV2Page from "@/modules/scores/ScoresV2Page";

describe("ScoresV2Page", () => {
  it("rend la page V2", () => {
    const { container, getByText } = render(<ScoresV2Page />);
    expect(getByText(/Scores \(V2\)/i)).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
