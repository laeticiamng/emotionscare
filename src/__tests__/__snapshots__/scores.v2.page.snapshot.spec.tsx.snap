// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ScoresV2Page > rend la page V2 1`] = `
<div>
  <main aria-label="Scores V2">
    <header>
      <h1>Scores (V2)</h1>
      <p>Progression, streaks, badges et historique</p>
    </header>
    <div>
      <div>ScoresV2Panel</div>
      <div style="margin-top: 12px">
        <a href="/modules/scores">Revenir Ã  la version actuelle</a>
      </div>
    </div>
  </main>
</div>
`;
